addon/validations/factory.js

```diff
@@ -153,68 +159,105 @@ function processDefaultOptions(validations = {}) {
 }
 
 /**
- * Creates the validations object that will become `model.validations`. Creates all necessary global CPs such as
- * `isValid`, `isAsync`, etc and also the `attrs` object.
- * @method createValidationsObject
+ * Creates the validations class that will become `model.validations`.
+ *   - Setup parent validation inheritance
+ *   - Normalize nested keys (i.e. 'details.dob') into objects (i.e { details: { dob: validator() }})
+ *   - Merge normalized validations with parent
+ *   - Create `attrs` object with CPs
+ *   - Create global CPs (i.e. 'isValid', 'messages', etc...)
+ *
+ * @method createValidationsClass
  * @private
  * @param  {Object} validations
  * @return {Ember.Object}
  */
-function createValidationsObject(validations = {}) {
-  return Ember.Object.extend({
+function createValidationsClass(inheritedValidations, validations = {}) {
+  let validationRules = {};
+  let validatableAttributes = Object.keys(validations);
+
+  // Setup validation inheritance
+  if(inheritedValidations) {
+    validationRules = merge(validationRules, inheritedValidations.get('_validationRules'));
+    validatableAttributes = emberArray(inheritedValidations.get('validatableAttributes').concat(validatableAttributes)).uniq();
+  }
+
+  // Normalize nested keys into actual objects and merge them with parent object
+  Object.keys(validations).reduce((obj, key) => {
+    assign(obj, key, validations[key]);
+    return obj;
+  }, validationRules);
+
+  // Create the CPs that will be a part of the `attrs` object
+  let attrCPs = validatableAttributes.reduce((obj, attribute) => {
+    assign(obj, attribute, createCPValidationFor(attribute, validationRules[attribute]), true);
+    return obj;
+  }, {});
+
+  // Create the mixin that holds all the top level validation props (isValid, messages, etc)
+  const TopLevelProps = createTopLevelPropsMixin(validatableAttributes);
+
+  // Create the `attrs` class which will add the current model reference once instantiated
+  const Attrs = Ember.Object.extend(attrCPs, {
+    init() {
+      this._super(...arguments);
+      let model = this.get('_model');
+
+      validatableAttributes.forEach((attribute) => {
+        // Add a reference to the model in the deepest object
+        let path = attribute.split('.');
+        let lastObject = get(this, path.slice(0, path.length  - 1).join('.'));
+        if(isNone(get(lastObject, '_model'))) {
```

Comment:
if you have `foo.bar.baz` and `foo.bar.blah`, then it will prevent setting the model twice on `bar` but really, it doesnt need to be there.